src/Nethermind/Nethermind.Optimism/CL/ExecutionEngineManager.cs

```diff
@@ -138,4 +215,11 @@ private async Task<bool> SendForkChoiceUpdated(
 
         return true;
     }
+    public Task<ulong?> GetCurrentFinalizedBlockNumber()
+    {
+        return Task.FromResult(_currentFinalizedHead.Number != 0 ? _currentFinalizedHead.Number : (ulong?)null);
+    }
```

Comment:
Potential data race when accessing `_currentFinalizedHead`?

```suggestion
    public Task<ulong?> GetCurrentFinalizedBlockNumber()
    {
      await _semaphore.WaitAsync();
      try {
          return _currentFinalizedHead.Number != 0 ? _currentFinalizedHead.Number : (ulong?)null;
      } finally
      {
          _semaphore.Release();
      }
    }
```